insalan/payment/hooks.py

```diff
@@ -0,0 +1,116 @@
+"""
+Hooks module for payment
+
+This file is not part of the Django rest framework. Rather, it is a component
+designed for other components of the application to come and register custom
+handlers for payment success or failure.
+"""
+
+import logging
+
+from django.utils.translation import gettext_lazy as _
+
+from .models import ProductCategory
+
+
+class PaymentCallbackSystem:
+    """
+    Interface to register your hooks with, and get them
+    """
+
+    # The dictionary of hooks
+    __HOOKS = {}
+    __logger = logging.getLogger("insalan.payment.hooks.PaymentCallbackSystem")
+
+    @classmethod
+    def register_handler(cls, prodcat, handler, overwrite=False):
+        """
+        Register a handler for a product category
+
+        You give it a product (any instance of
+        `payment.models.ProductCategory`), and a handler (any subclass of
+        `payment.hooks.PaymentHooks`).
+
+        Overwriting an existing hook will trigger an error, unless you set
+        `overwrite=False`.
+        """
+
+        # Verify arguments
+        # 1. product is an instance of Product
+        if not isinstance(prodcat, ProductCategory):
+            raise ValueError(_("Produit qui n'est pas une instance de Produit"))
+
+        if not issubclass(handler, PaymentHooks):
+            raise ValueError(_("Descripteur qui ne dérive pas de PaymentHooks"))
+
+        if cls.__HOOKS.get(prodcat):
+            if overwrite:
+                cls.__logger.warning("Overwriting handler for product category %s", prodcat)
```

Comment:
f-string is more a more modern syntax, and as it used across the code base, we may switch to f-strings

Reply:
Here it is a syntax from the logger class (that probably calls `.format()` internally), I don't think this is a big deal. However, the content could be translated? We didn't agree upon a standard for internal logging

Reply:
Pylint and other linters recommend not formatting the strings, as it may be preferable to delay formatting until the very end inside of the `logger` methods, this is why I use these things here.

As for translation, because I document everything in English, I consider that everything that is meant for internal use is in English. We never agreed on it, however, so we could probably discuss it later.

Reply:
I agree that using English for internal logging and documentation seems easier as it is closer to the code and will avoid mix-ups between French and English words